tests/hardware/plugins/decades/test_decades.py

```diff
@@ -38,49 +36,61 @@ def test_close(decades: Decades) -> None:
 
 
 @patch("json.loads")
-@patch("finesse.hardware.plugins.decades.decades.get_decades_data")
+@patch("finesse.hardware.plugins.decades.decades.Decades._get_decades_data")
 def test_on_reply_received_no_error(
-    get_decades_data_mock: Mock, json_loads_mock: Mock, qtbot
+    _get_decades_data_mock: Mock, json_loads_mock: Mock, decades: Decades, qtbot
 ) -> None:
     """Test the _on_reply_received() method works when no error occurs."""
     reply = MagicMock()
     reply.error.return_value = QNetworkReply.NetworkError.NoError
 
     # NB: This value is of the wrong type, but it doesn't matter for here
-    get_decades_data_mock.return_value = "cool_sensor_data"
+    _get_decades_data_mock.return_value = "cool_sensor_data"
 
     # Check the correct pubsub message is sent
-    assert _on_reply_received(reply) == "cool_sensor_data"
+    assert decades._on_reply_received(reply) == "cool_sensor_data"
     json_loads_mock.assert_called_once_with(reply.readAll().data().decode())
 
 
-def test_on_reply_received_network_error(qtbot) -> None:
+def test_on_reply_received_network_error(decades: Decades, qtbot) -> None:
     """Tests the _on_reply_received() method works when a network error occurs."""
     reply = MagicMock()
     reply.error.return_value = QNetworkReply.NetworkError.HostNotFoundError
     reply.errorString.return_value = "Host not found"
 
     with pytest.raises(DecadesError):
         # Check the correct pubsub message is sent
-        _on_reply_received(reply)
+        decades._on_reply_received(reply)
 
 
 @patch("json.loads")
-@patch("finesse.hardware.plugins.decades.decades.get_decades_data")
+@patch("finesse.hardware.plugins.decades.decades.Decades._get_decades_data")
 def test_on_reply_received_exception(
-    get_decades_data_mock: Mock, json_loads_mock: Mock, qtbot
+    _get_decades_data_mock: Mock, json_loads_mock: Mock, decades: Decades, qtbot
 ) -> None:
     """Tests the _on_reply_received() method works when an exception is raised."""
     reply = MagicMock()
     reply.error.return_value = QNetworkReply.NetworkError.NoError
 
     # Make get_decades_data() raise an exception
     error = Exception()
-    get_decades_data_mock.side_effect = error
+    _get_decades_data_mock.side_effect = error
 
     with pytest.raises(Exception):
         # Check the correct pubsub message is sent
-        _on_reply_received(reply)
+        decades._on_reply_received(reply)
+
+
+def test_send_params(decades: Decades, qtbot) -> None:
+    """Tests the send_data() method."""
+    with patch.object(decades, "_requester") as requester_mock:
+        with patch.object(decades, "pubsub_broadcast") as broadcast_mock:
+            broadcast_mock.return_value = "WRAPPED_FUNC"
+            decades.send_params()
+            broadcast_mock.assert_called_once_with(
+                decades._on_params_received, "params.response"
+            )
+            requester_mock.make_request.assert_called_once_with(ANY, "WRAPPED_FUNC")
```

Comment:
Ha, I didn't know about this `ANY` parameter. And I've done some nasty hacks elsewhere to work around this 🙃

Reply:
This was adapted from @jamesturner246's `test_send_data` test, so I take no credit 😅